src/steps/ExitPoint.js

```diff
@@ -1,15 +1,43 @@
 const Page = require('./Page');
 const destroySession = require('../session/destroySession');
 const { stopHere } = require('../flow');
+const deepEqual = require('deep-equal');
+
+storeStateInUrl = (req, res, next) => {
+  if (!deepEqual(req.query, req.currentStep.values())) {
+    const params = Object.entries(req.currentStep.values())
+      .map(([key, val]) => `${key}=${val}`)
+      .join('&');
+    res.redirect(`${req.route.path}?${params}`);
+  } else {
+    next();
+  }
+};
 
 class ExitPoint extends Page {
   get middleware() {
-    return [...super.middleware, destroySession];
+    if (this.req.session.active()) {
+      return [
+        this.journey.collectSteps,
```

Comment:
If we do the `this.journey.getField` thing I mentioned above then we wouldn't need to collectSteps here.